ui/ducks/swaps/swaps.js

```diff
@@ -1365,7 +1365,25 @@ export function fetchMetaSwapsGasPriceEstimates() {
       dispatch(swapGasPriceEstimatesFetchFailed());
 
       try {
-        const gasPrice = await global.ethQuery.gasPrice();
+        const gasPrice = await new Promise((resolve, reject) => {
+          global.ethereumProvider.sendAsync(
```

Comment:
Do we want to use `request` here?

Reply:
The current `global.ethereumProvider` is not EIP-1193 compliant, similar to the `global.ethQuery` that we are removing in this change. However, the goal of this update is to retain only `global.ethereumProvider`. Once this change is merged, `global.ethereumProvider` will be updated to use a stream provider that complies with EIP-1193, allowing us to utilize the request method. This update will be implemented in a subsequent PR, which is currently in draft: https://github.com/MetaMask/metamask-extension/pull/29650.

Reply:
Ah sorry I think I got confused with that PR. Thanks :)